{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6a765fdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import pprint\n",
    "import openpyxl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5ef87af0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['.DS_Store',\n",
       " '.ipynb_checkpoints',\n",
       " 'annotations.json',\n",
       " 'annotations_1.xlsx',\n",
       " 'annotations_10.xlsx',\n",
       " 'annotations_11.xlsx',\n",
       " 'annotations_12.xlsx',\n",
       " 'annotations_13.xlsx',\n",
       " 'annotations_14.xlsx',\n",
       " 'annotations_15.xlsx',\n",
       " 'annotations_16.xlsx',\n",
       " 'annotations_17.xlsx',\n",
       " 'annotations_18.xlsx',\n",
       " 'annotations_19.xlsx',\n",
       " 'annotations_2.xlsx',\n",
       " 'annotations_20.xlsx',\n",
       " 'annotations_21.xlsx',\n",
       " 'annotations_22.xlsx',\n",
       " 'annotations_3.xlsx',\n",
       " 'annotations_4.xlsx',\n",
       " 'annotations_5.xlsx',\n",
       " 'annotations_6.xlsx',\n",
       " 'annotations_7.xlsx',\n",
       " 'annotations_8.xlsx',\n",
       " 'annotations_9.xlsx',\n",
       " 'merged_data.xlsx',\n",
       " 'Merger.ipynb',\n",
       " 'Regiatery_Complete.csv',\n",
       " 'submatcheds.txt',\n",
       " 'Untitled.ipynb',\n",
       " 'Untitled1 - Jupyter Notebook.pdf',\n",
       " 'Untitled1.ipynb']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9ea0336",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "af25f997",
   "metadata": {},
   "outputs": [],
   "source": [
    "registry = pd.read_csv(\"Regiatery_Complete.csv\", low_memory=False)\n",
    "softcite = pd.read_excel(\"merged_data.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "974bb923",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rid</th>\n",
       "      <th>scr_id</th>\n",
       "      <th>original_id</th>\n",
       "      <th>type</th>\n",
       "      <th>parent_organization_id</th>\n",
       "      <th>Resource_Name</th>\n",
       "      <th>Defining_Citation</th>\n",
       "      <th>Supercategory</th>\n",
       "      <th>Species</th>\n",
       "      <th>Related_Disease</th>\n",
       "      <th>...</th>\n",
       "      <th>PubMed_Articles</th>\n",
       "      <th>Intellectual_Property</th>\n",
       "      <th>Patient_Engagement</th>\n",
       "      <th>Geo_Location</th>\n",
       "      <th>Relation</th>\n",
       "      <th>Canonical_ID</th>\n",
       "      <th>relationship_ids</th>\n",
       "      <th>relationship_strings</th>\n",
       "      <th>resources_scr_ids</th>\n",
       "      <th>resources_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>SCR_000001</td>\n",
       "      <td>nlx_152482</td>\n",
       "      <td>Organization</td>\n",
       "      <td>(null)</td>\n",
       "      <td>TransGenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Organization</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SCR_000001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>SCR_000002</td>\n",
       "      <td>nlx_152901</td>\n",
       "      <td>Resource</td>\n",
       "      <td>SCR_001373</td>\n",
       "      <td>monarch-ontologies</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Resource</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SCR_000002</td>\n",
       "      <td>6|7</td>\n",
       "      <td>is_related_to|has_parent_organization</td>\n",
       "      <td>SCR_003214|SCR_001373</td>\n",
       "      <td>Vertebrate Trait Ontology|Monarch Initiative</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>SCR_000003</td>\n",
       "      <td>nlx_158000</td>\n",
       "      <td>Resource</td>\n",
       "      <td>(null)</td>\n",
       "      <td>Sarah Cannon Research Institute; Tennessee; USA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SCR_000003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>SCR_000004</td>\n",
       "      <td>nlx_152368</td>\n",
       "      <td>Organization</td>\n",
       "      <td>(null)</td>\n",
       "      <td>GE Healthcare</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Organization</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SCR_000004</td>\n",
       "      <td>4|7</td>\n",
       "      <td>lists|is_parent_organization_of</td>\n",
       "      <td>SCR_013566|SCR_014246</td>\n",
       "      <td>Amersham Biosciences|ImageQuant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>SCR_000005</td>\n",
       "      <td>nif-0000-00023</td>\n",
       "      <td>Resource</td>\n",
       "      <td>(null)</td>\n",
       "      <td>Neuroshare - Open data specifications and soft...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Resource</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SCR_000005</td>\n",
       "      <td>6</td>\n",
       "      <td>is_related_to</td>\n",
       "      <td>SCR_004951</td>\n",
       "      <td>BrainLiner</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 51 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   rid      scr_id     original_id          type parent_organization_id  \\\n",
       "0    1  SCR_000001      nlx_152482  Organization                 (null)   \n",
       "1    2  SCR_000002      nlx_152901      Resource             SCR_001373   \n",
       "2    3  SCR_000003      nlx_158000      Resource                 (null)   \n",
       "3    4  SCR_000004      nlx_152368  Organization                 (null)   \n",
       "4    5  SCR_000005  nif-0000-00023      Resource                 (null)   \n",
       "\n",
       "                                       Resource_Name Defining_Citation  \\\n",
       "0                                         TransGenic               NaN   \n",
       "1                                 monarch-ontologies               NaN   \n",
       "2    Sarah Cannon Research Institute; Tennessee; USA               NaN   \n",
       "3                                      GE Healthcare               NaN   \n",
       "4  Neuroshare - Open data specifications and soft...               NaN   \n",
       "\n",
       "  Supercategory Species Related_Disease  ... PubMed_Articles  \\\n",
       "0  Organization     NaN             NaN  ...             NaN   \n",
       "1      Resource     NaN             NaN  ...             NaN   \n",
       "2           NaN     NaN             NaN  ...             NaN   \n",
       "3  Organization     NaN             NaN  ...             NaN   \n",
       "4      Resource     NaN             NaN  ...             NaN   \n",
       "\n",
       "  Intellectual_Property Patient_Engagement Geo_Location Relation Canonical_ID  \\\n",
       "0                   NaN                NaN          NaN      NaN   SCR_000001   \n",
       "1                   NaN                NaN          NaN      NaN   SCR_000002   \n",
       "2                   NaN                NaN          NaN      NaN   SCR_000003   \n",
       "3                   NaN                NaN          NaN      NaN   SCR_000004   \n",
       "4                   NaN                NaN          NaN      NaN   SCR_000005   \n",
       "\n",
       "  relationship_ids                   relationship_strings  \\\n",
       "0              NaN                                    NaN   \n",
       "1              6|7  is_related_to|has_parent_organization   \n",
       "2              NaN                                    NaN   \n",
       "3              4|7        lists|is_parent_organization_of   \n",
       "4                6                          is_related_to   \n",
       "\n",
       "       resources_scr_ids                               resources_names  \n",
       "0                    NaN                                           NaN  \n",
       "1  SCR_003214|SCR_001373  Vertebrate Trait Ontology|Monarch Initiative  \n",
       "2                    NaN                                           NaN  \n",
       "3  SCR_013566|SCR_014246               Amersham Biosciences|ImageQuant  \n",
       "4             SCR_004951                                    BrainLiner  \n",
       "\n",
       "[5 rows x 51 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "registry.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "25842d3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "softcite_Name = []\n",
    "for i in list(softcite[\"sn\"]):\n",
    "    if type(i) == str:\n",
    "        softcite_Name.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "92e8c3a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "registry_Name = []\n",
    "source_Name = []\n",
    "for i in list(registry[\"Resource_Name\"]):\n",
    "    if type(i) == str:\n",
    "        registry_Name.append(i)\n",
    "        \n",
    "for i in list(registry[\"resources_names\"]):\n",
    "    if type(i) == str:\n",
    "        source_Name.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "16f6393d",
   "metadata": {},
   "outputs": [],
   "source": [
    "matching = []\n",
    "submatch = []\n",
    "for i in softcite_Name:\n",
    "    for j in registry_Name:\n",
    "        if i == j:\n",
    "            matching.append(i)\n",
    "        if i in j:\n",
    "            submatch.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d5f1f876",
   "metadata": {},
   "outputs": [],
   "source": [
    "resourcematch = []\n",
    "resmatchedSUB = []\n",
    "\n",
    "for i in softcite_Name:\n",
    "    for j in source_Name:\n",
    "        if i == j:\n",
    "            resourcematch.append(i)\n",
    "        if i in j:\n",
    "            resmatchedSUB.append(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "20412382",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SPSS',\n",
       " 'SPSS',\n",
       " 'SPSS',\n",
       " 'Skyline',\n",
       " 'Skyline',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'SAS',\n",
       " 'Covidence',\n",
       " 'COMPASS',\n",
       " 'COMPASS',\n",
       " 'COMPASS',\n",
       " 'PHYLIP',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Excel',\n",
       " 'Prism',\n",
       " 'Prism',\n",
       " 'Prism',\n",
       " 'Prism',\n",
       " 'Prism',\n",
       " 'ImmunoSpot',\n",
       " 'FlowJo',\n",
       " 'FlowJo',\n",
       " 'Adobe Illustrator',\n",
       " 'FACSDiva',\n",
       " 'Aperio ImageScope',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'Ensembl',\n",
       " 'FACSJazz',\n",
       " 'FACS Sortware',\n",
       " 'StepOnePlus',\n",
       " 'MoFlo Astrios',\n",
       " 'CytExpert',\n",
       " 'biomaRt',\n",
       " 'Summit',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Google',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'Ger',\n",
       " 'JMP Pro',\n",
       " 'G*Power',\n",
       " 'MATLAB',\n",
       " 'MATLAB',\n",
       " 'MATLAB',\n",
       " 'MATLAB',\n",
       " 'MATLAB',\n",
       " 'MATLAB',\n",
       " 'SPSS Statistics',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'BLAST',\n",
       " 'Windows',\n",
       " 'Windows',\n",
       " 'Windows',\n",
       " 'ClustalW',\n",
       " 'Clustal W',\n",
       " 'MEGA',\n",
       " 'MEGA',\n",
       " 'MEGA',\n",
       " 'MEGA',\n",
       " 'MEGA',\n",
       " 'MEGA',\n",
       " 'MEGA',\n",
       " 'AMBER',\n",
       " 'AMBER',\n",
       " 'AMBER',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Tracer',\n",
       " 'Geneious',\n",
       " 'Geneious',\n",
       " 'MAFFT',\n",
       " 'jModelTest',\n",
       " 'NAMD',\n",
       " 'TASSER',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'Python',\n",
       " 'STATISTICA',\n",
       " 'GraphPad Prism',\n",
       " 'SearchGUI',\n",
       " 'MaxQuant',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'LAST',\n",
       " 'BWA',\n",
       " 'BWA',\n",
       " 'FastQC',\n",
       " 'OMSSA',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'STAR',\n",
       " 'Trimmomatic',\n",
       " 'RSEM',\n",
       " 'Mascot',\n",
       " 'Mascot',\n",
       " 'Synapse',\n",
       " 'Synapse',\n",
       " 'Synapse',\n",
       " 'Synapse',\n",
       " 'Synapse',\n",
       " 'Synapse',\n",
       " 'Synapse',\n",
       " 'GitLab',\n",
       " 'ImageJ',\n",
       " 'ImageJ',\n",
       " 'ImageJ',\n",
       " 'ImageJ',\n",
       " 'AmberTools',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'script',\n",
       " 'Google Scholar',\n",
       " 'Google Scholar',\n",
       " 'Minitab',\n",
       " 'Maestro',\n",
       " 'Maestro',\n",
       " 'NVivo',\n",
       " 'SYBYL',\n",
       " 'ggplot',\n",
       " 'PhyML',\n",
       " 'DIAMOND',\n",
       " 'DIAMOND',\n",
       " 'Primer Express',\n",
       " 'Primer Express',\n",
       " 'SPAdes',\n",
       " 'SPAdes',\n",
       " 'ASTRA',\n",
       " 'ASTRA',\n",
       " 'GENESIS',\n",
       " 'GENESIS',\n",
       " 'GENESIS',\n",
       " 'Picard',\n",
       " 'Picard',\n",
       " 'Picard',\n",
       " 'ClueGO',\n",
       " 'Cytoscape',\n",
       " 'Cytoscape',\n",
       " 'Cytoscape',\n",
       " 'MCODE',\n",
       " 'PaintOmics',\n",
       " 'CluePedia',\n",
       " 'miRanda',\n",
       " 'PITA',\n",
       " 'TargetScan',\n",
       " 'miRWalk',\n",
       " 'JMP',\n",
       " 'JMP',\n",
       " 'JMP',\n",
       " 'IDEA',\n",
       " 'IDEA',\n",
       " 'GTEx',\n",
       " 'GTEx',\n",
       " 'Olink',\n",
       " 'ChEMBL',\n",
       " 'SNPTEST',\n",
       " 'METAL',\n",
       " 'METAL',\n",
       " 'TwoSampleMR',\n",
       " 'PLINK',\n",
       " 'PLINK',\n",
       " 'PLINK',\n",
       " 'PLINK',\n",
       " 'PLINK',\n",
       " 'MetaboAnalyst',\n",
       " 'MetaboAnalyst',\n",
       " 'EpiModel',\n",
       " 'EpiModel',\n",
       " 'EpiModel',\n",
       " 'SPSS Statistic',\n",
       " 'Magellan',\n",
       " 'RZ',\n",
       " 'RZ',\n",
       " 'BioSig',\n",
       " 'BioSig',\n",
       " 'BioSig',\n",
       " 'Fastp',\n",
       " 'bedtools',\n",
       " 'FCS Express',\n",
       " 'Phoenix',\n",
       " 'Phoenix',\n",
       " 'Phoenix',\n",
       " 'Phoenix',\n",
       " 'Phoenix',\n",
       " 'Orange',\n",
       " 'APT',\n",
       " 'APT',\n",
       " 'APT',\n",
       " 'SNP-sites',\n",
       " 'Guppy',\n",
       " 'BEAGLE',\n",
       " 'BEAGLE',\n",
       " 'EMBOSS',\n",
       " 'EMBOSS',\n",
       " 'EMBOSS',\n",
       " 'cBioPortal',\n",
       " 'DiCon',\n",
       " 'FRED',\n",
       " 'FRED',\n",
       " 'Image Studio Lite',\n",
       " 'SeqMan',\n",
       " 'SoftMax Pro',\n",
       " 'MyHits',\n",
       " 'ArcGIS',\n",
       " 'IDRISI',\n",
       " 'REDCap',\n",
       " 'SOAPsnp',\n",
       " 'MOE',\n",
       " 'GATK',\n",
       " 'PyMOL',\n",
       " 'PyMOL',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Stan',\n",
       " 'Neurolucida',\n",
       " 'Neurolucida',\n",
       " 'Neurolucida',\n",
       " 'MetaMorph',\n",
       " 'Photoshop',\n",
       " 'Photoshop',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'R package',\n",
       " 'Linux',\n",
       " 'Linux',\n",
       " 'Linux',\n",
       " 'AliView',\n",
       " 'Origin',\n",
       " 'Origin',\n",
       " 'Origin',\n",
       " 'Origin',\n",
       " 'Origin',\n",
       " 'Origin',\n",
       " 'Refmac',\n",
       " 'CCP4',\n",
       " 'CCP4',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'CNS',\n",
       " 'ProtParam',\n",
       " 'Zoom',\n",
       " 'Zoom',\n",
       " 'Zoom',\n",
       " 'Zoom',\n",
       " 'Zoom',\n",
       " 'Zoom',\n",
       " 'VaxiJen',\n",
       " 'HADDOCK',\n",
       " 'ToxinPred',\n",
       " 'PROCHECK',\n",
       " 'PyRx',\n",
       " 'GOLD',\n",
       " 'GOLD',\n",
       " 'XLSTAT',\n",
       " 'YASARA',\n",
       " 'GAMM',\n",
       " 'iOS',\n",
       " 'iOS',\n",
       " 'Spark',\n",
       " 'Spark',\n",
       " 'Spark',\n",
       " 'Spark',\n",
       " 'Spark',\n",
       " 'spdep',\n",
       " 'Scala',\n",
       " 'Scala',\n",
       " 'Scala',\n",
       " 'Scala',\n",
       " 'Scala',\n",
       " 'Hadoop',\n",
       " 'Hadoop',\n",
       " 'Apache',\n",
       " 'Apache',\n",
       " 'Apache',\n",
       " 'Apache Spark',\n",
       " 'Apache Hadoop',\n",
       " 'Bioconductor',\n",
       " 'Bioconductor',\n",
       " 'Basic Local Alignment Search Tool',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'Scholar',\n",
       " 'SynergyFinder',\n",
       " 'Prokka',\n",
       " 'SWISS-MODEL',\n",
       " 'SWISS-MODEL',\n",
       " 'PDBsum',\n",
       " 'ModBase',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'Swiss',\n",
       " 'MODELLER',\n",
       " 'ITOL',\n",
       " 'PubMed',\n",
       " 'PubMed',\n",
       " 'PubMed',\n",
       " 'PubMed',\n",
       " 'PubMed',\n",
       " 'PubMed',\n",
       " 'Gwyddion',\n",
       " 'IMPUTE',\n",
       " 'IMPUTE',\n",
       " 'IMPUTE',\n",
       " 'Azure',\n",
       " 'SDS',\n",
       " 'SDS',\n",
       " 'Ceph',\n",
       " 'Ceph',\n",
       " 'SQL',\n",
       " 'SQL',\n",
       " 'SQL',\n",
       " 'AcqKnowledge',\n",
       " 'EpiData',\n",
       " 'scATAC',\n",
       " 'scATAC',\n",
       " 'scVelo',\n",
       " 'ComplexHeatmap',\n",
       " 'ArchR',\n",
       " 'AIMLESS',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'Imaging and',\n",
       " 'UMAP',\n",
       " 'UMAP',\n",
       " 'Phenograph',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'HT',\n",
       " 'UCSF Chimera',\n",
       " 'UCSF Chimera',\n",
       " 'RELION',\n",
       " 'Rosetta',\n",
       " 'Rosetta',\n",
       " 'Rosetta',\n",
       " 'Rosetta',\n",
       " 'Rosetta',\n",
       " 'Rosetta',\n",
       " 'Rosetta',\n",
       " 'Mesquite',\n",
       " 'PAUP',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'YouTube',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'COVID-',\n",
       " 'Julia',\n",
       " 'Julia',\n",
       " 'DESeq',\n",
       " 'DESeq',\n",
       " 'DESeq2',\n",
       " 'Image Lab',\n",
       " 'Clustal Omega',\n",
       " 'CFX Manager',\n",
       " 'Translate',\n",
       " 'GISAID',\n",
       " 'GROMACS',\n",
       " 'EndNote',\n",
       " 'GeneSpring GX',\n",
       " 'Image-Pro Plus',\n",
       " 'Image-Pro Plus',\n",
       " 'Harmony',\n",
       " 'Harmony',\n",
       " 'Nextstrain',\n",
       " 'GeneTools',\n",
       " 'GeneTools',\n",
       " 'CellQuest',\n",
       " 'RDP',\n",
       " 'RDP',\n",
       " 'RDP',\n",
       " 'LigPlot',\n",
       " 'Pandas',\n",
       " 'NumPy',\n",
       " 'SciPy',\n",
       " 'Excel Statistics',\n",
       " 'BLASTP',\n",
       " 'BLASTP',\n",
       " 'Pfam',\n",
       " 'SignalP',\n",
       " 'REViGO',\n",
       " 'TMHMM',\n",
       " 'Primer-BLAST',\n",
       " 'Trinity',\n",
       " 'Trinity',\n",
       " 'Trinity',\n",
       " 'Trinity',\n",
       " 'Trinity',\n",
       " 'BLASTX',\n",
       " 'BLASTX',\n",
       " 'EnrichmentMap',\n",
       " 'NOVA',\n",
       " 'NOVA',\n",
       " 'NOVA',\n",
       " 'NOVA',\n",
       " 'NOVA',\n",
       " 'swMath',\n",
       " 'WinBUGS',\n",
       " 'ggplot2',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'car',\n",
       " 'PerlPrimer',\n",
       " 'Chimera',\n",
       " 'Chimera',\n",
       " 'Chimera',\n",
       " 'Chimera',\n",
       " 'Chimera',\n",
       " 'R script',\n",
       " 'VMD',\n",
       " 'Glide',\n",
       " 'ORA',\n",
       " 'ORA',\n",
       " 'ORA',\n",
       " 'ORA',\n",
       " 'ORA',\n",
       " 'ORA',\n",
       " 'ORA',\n",
       " 'Swarm',\n",
       " 'R Project',\n",
       " 'sna',\n",
       " 'sna',\n",
       " 'sna',\n",
       " 'igraph',\n",
       " 'igraph',\n",
       " 'RStudio',\n",
       " 'dplyr',\n",
       " 'CyTOF',\n",
       " 'CyTOF',\n",
       " 'CyTOF',\n",
       " 'openxlsx',\n",
       " 'tidyr',\n",
       " 'Enrichr',\n",
       " 'HISAT2',\n",
       " 'mixOmics',\n",
       " 'featureCounts',\n",
       " 'EdgeR',\n",
       " 'GalaxyRefine',\n",
       " 'SCRATCH',\n",
       " 'SnapGene',\n",
       " 'SnapGene',\n",
       " 'Living Image',\n",
       " 'Gen5',\n",
       " 'Nanopolish',\n",
       " ...]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "submatch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a8b54f43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SPSS',\n",
       " 'Skyline',\n",
       " 'Covidence',\n",
       " 'COMPASS',\n",
       " 'COMPASS',\n",
       " 'COMPASS',\n",
       " 'PHYLIP',\n",
       " 'FlowJo',\n",
       " 'Adobe Illustrator',\n",
       " 'Aperio ImageScope',\n",
       " 'Ensembl',\n",
       " 'biomaRt',\n",
       " 'Google',\n",
       " 'G*Power',\n",
       " 'MATLAB',\n",
       " 'ClustalW',\n",
       " 'AMBER',\n",
       " 'Tracer',\n",
       " 'Geneious',\n",
       " 'MAFFT',\n",
       " 'jModelTest',\n",
       " 'NAMD',\n",
       " 'GraphPad Prism',\n",
       " 'SearchGUI',\n",
       " 'MaxQuant',\n",
       " 'LAST',\n",
       " 'BWA',\n",
       " 'FastQC',\n",
       " 'STAR',\n",
       " 'Trimmomatic',\n",
       " 'RSEM',\n",
       " 'Mascot',\n",
       " 'Synapse',\n",
       " 'GitLab',\n",
       " 'ImageJ',\n",
       " 'AmberTools',\n",
       " 'Google Scholar',\n",
       " 'Minitab',\n",
       " 'Maestro',\n",
       " 'NVivo',\n",
       " 'PhyML',\n",
       " 'DIAMOND',\n",
       " 'DIAMOND',\n",
       " 'Primer Express',\n",
       " 'SPAdes',\n",
       " 'ASTRA',\n",
       " 'Picard',\n",
       " 'ClueGO',\n",
       " 'Cytoscape',\n",
       " 'MCODE',\n",
       " 'PaintOmics',\n",
       " 'miRanda',\n",
       " 'PITA',\n",
       " 'TargetScan',\n",
       " 'miRWalk',\n",
       " 'JMP',\n",
       " 'ChEMBL',\n",
       " 'SNPTEST',\n",
       " 'METAL',\n",
       " 'TwoSampleMR',\n",
       " 'PLINK',\n",
       " 'MetaboAnalyst',\n",
       " 'EpiModel',\n",
       " 'FCS Express',\n",
       " 'Phoenix',\n",
       " 'SNP-sites',\n",
       " 'BEAGLE',\n",
       " 'EMBOSS',\n",
       " 'cBioPortal',\n",
       " 'Image Studio Lite',\n",
       " 'MyHits',\n",
       " 'IDRISI',\n",
       " 'REDCap',\n",
       " 'SOAPsnp',\n",
       " 'MOE',\n",
       " 'GATK',\n",
       " 'PyMOL',\n",
       " 'Stan',\n",
       " 'Neurolucida',\n",
       " 'AliView',\n",
       " 'Origin',\n",
       " 'Refmac',\n",
       " 'CCP4',\n",
       " 'VaxiJen',\n",
       " 'HADDOCK',\n",
       " 'ToxinPred',\n",
       " 'PROCHECK',\n",
       " 'PyRx',\n",
       " 'GOLD',\n",
       " 'XLSTAT',\n",
       " 'YASARA',\n",
       " 'Spark',\n",
       " 'spdep',\n",
       " 'Apache Spark',\n",
       " 'Apache Hadoop',\n",
       " 'Bioconductor',\n",
       " 'SynergyFinder',\n",
       " 'Prokka',\n",
       " 'SWISS-MODEL',\n",
       " 'PDBsum',\n",
       " 'ModBase',\n",
       " 'MODELLER',\n",
       " 'PubMed',\n",
       " 'Gwyddion',\n",
       " 'IMPUTE',\n",
       " 'scVelo',\n",
       " 'ComplexHeatmap',\n",
       " 'ArchR',\n",
       " 'AIMLESS',\n",
       " 'Phenograph',\n",
       " 'UCSF Chimera',\n",
       " 'RELION',\n",
       " 'Rosetta',\n",
       " 'Mesquite',\n",
       " 'PAUP',\n",
       " 'DESeq',\n",
       " 'DESeq2',\n",
       " 'Clustal Omega',\n",
       " 'CFX Manager',\n",
       " 'GISAID',\n",
       " 'GROMACS',\n",
       " 'EndNote',\n",
       " 'GeneSpring GX',\n",
       " 'Image-Pro Plus',\n",
       " 'Image-Pro Plus',\n",
       " 'Harmony',\n",
       " 'Harmony',\n",
       " 'Nextstrain',\n",
       " 'GeneTools',\n",
       " 'GeneTools',\n",
       " 'Pandas',\n",
       " 'NumPy',\n",
       " 'SciPy',\n",
       " 'Excel Statistics',\n",
       " 'BLASTP',\n",
       " 'Pfam',\n",
       " 'SignalP',\n",
       " 'REViGO',\n",
       " 'Primer-BLAST',\n",
       " 'Trinity',\n",
       " 'BLASTX',\n",
       " 'EnrichmentMap',\n",
       " 'swMath',\n",
       " 'WinBUGS',\n",
       " 'ggplot2',\n",
       " 'PerlPrimer',\n",
       " 'Chimera',\n",
       " 'VMD',\n",
       " 'Glide',\n",
       " 'igraph',\n",
       " 'RStudio',\n",
       " 'dplyr',\n",
       " 'openxlsx',\n",
       " 'tidyr',\n",
       " 'Enrichr',\n",
       " 'HISAT2',\n",
       " 'mixOmics',\n",
       " 'featureCounts',\n",
       " 'GalaxyRefine',\n",
       " 'SCRATCH',\n",
       " 'SnapGene',\n",
       " 'Gen5',\n",
       " 'Nanopolish',\n",
       " 'pheatmap',\n",
       " 'FASTA',\n",
       " 'MyAssays',\n",
       " 'KEGG',\n",
       " 'siDirect',\n",
       " 'Integrative Genomics Viewer',\n",
       " 'HALO',\n",
       " 'FreeBayes',\n",
       " 'TempEst',\n",
       " 'karyoploteR',\n",
       " 'IQ-TREE',\n",
       " 'jamovi',\n",
       " 'HKL 3000',\n",
       " 'Django',\n",
       " 'Gephi',\n",
       " 'Kaluza',\n",
       " 'MEGAN',\n",
       " 'PRINSEQ',\n",
       " 'Entrez',\n",
       " 'Batch Entrez',\n",
       " 'cryoSPARC',\n",
       " 'Weka',\n",
       " 'Adobe Photoshop',\n",
       " 'IMOD',\n",
       " 'ImageMagick',\n",
       " 'OpenCV',\n",
       " 'Maple',\n",
       " 'Scilab',\n",
       " 'GenePix Pro',\n",
       " 'PSIPRED',\n",
       " 'KOBAS',\n",
       " 'GeoDa',\n",
       " 'DMLE',\n",
       " 'Blender',\n",
       " 'Jupyter Notebook',\n",
       " 'CalcuSyn',\n",
       " 'microRNA.org',\n",
       " 'DOCK',\n",
       " 'jMRUI',\n",
       " 'SC3',\n",
       " 'DAVID',\n",
       " 'CD-HIT',\n",
       " 'scikit-learn',\n",
       " 'PostgreSQL',\n",
       " 'NetOGlyc',\n",
       " 'CHARMM',\n",
       " 'Signal',\n",
       " 'Signal',\n",
       " 'Autogrid',\n",
       " 'ImageQuant',\n",
       " 'ggpubr',\n",
       " 'RAxML',\n",
       " 'afex',\n",
       " 'Snakemake',\n",
       " 'Galaxy',\n",
       " 'AMOS',\n",
       " 'HTSeq',\n",
       " 'Proteome Discoverer',\n",
       " 'GRADEpro',\n",
       " 'Statistical Analysis System',\n",
       " 'TBLASTX',\n",
       " 'InterProScan',\n",
       " 'Molecular Signatures Database',\n",
       " 'QuPath',\n",
       " 'ANTIGENpro',\n",
       " 'RAMPAGE',\n",
       " 'MolProbity',\n",
       " 'ProSA-web',\n",
       " 'RNAstructure',\n",
       " 'GingerALE',\n",
       " 'Mango',\n",
       " 'scran',\n",
       " 'ggtree',\n",
       " 'FastTree',\n",
       " 'VDJ',\n",
       " 'WordPress',\n",
       " 'COMSOL Multiphysics',\n",
       " 'Biorender',\n",
       " 'Avogadro',\n",
       " 'glmmADMB',\n",
       " 'SerialEM',\n",
       " 'Conda',\n",
       " 'LightGBM',\n",
       " 'psych',\n",
       " 'Jalview',\n",
       " 'Jpred',\n",
       " 'Progenesis QI',\n",
       " 'ANNOVAR',\n",
       " 'htseq-count',\n",
       " 'PubChem',\n",
       " 'FSL',\n",
       " 'CLC Genomics Workbench',\n",
       " 'MZmine',\n",
       " 'MultiQC',\n",
       " 'SQLite',\n",
       " 'SigmaPlot',\n",
       " 'Trim Galore',\n",
       " 'Porechop',\n",
       " 'Molinspiration',\n",
       " 'Phenix.refine',\n",
       " 'PROSITE',\n",
       " 'survminer',\n",
       " 'MuTect',\n",
       " 'clusterProfiler',\n",
       " 'Blast2GO',\n",
       " 'SmartPLS',\n",
       " 'SortMeRNA',\n",
       " 'PCAGO',\n",
       " 'Gnuplot',\n",
       " 'Epik',\n",
       " 'TOMCAT',\n",
       " 'Recombination Detection Program',\n",
       " 'ggrepel',\n",
       " 'ggraph',\n",
       " 'CONN',\n",
       " 'SPM',\n",
       " 'M3',\n",
       " 'KMC',\n",
       " 'GENCODE',\n",
       " 'XGBoost',\n",
       " 'emmeans',\n",
       " 'MetaXpress',\n",
       " 'Rayyan QCRI',\n",
       " 'Pathway Commons',\n",
       " 'STRING',\n",
       " 'QIIME2',\n",
       " 'NetworkX',\n",
       " 'Simulx',\n",
       " 'iTasser',\n",
       " 'Praat',\n",
       " 'Open Babel',\n",
       " 'SleepSign',\n",
       " 'BUSTER',\n",
       " 'CheckMyMetal',\n",
       " 'Discovery Studio Visualizer',\n",
       " 'Cytobank',\n",
       " 'survival',\n",
       " 'JASP',\n",
       " 'GEO2R',\n",
       " 'cytoHubba',\n",
       " 'COVID CG',\n",
       " 'miRDB',\n",
       " 'RNAhybrid',\n",
       " 'RNA22',\n",
       " 'EMAN',\n",
       " 'NetMHC',\n",
       " 'ProFunc',\n",
       " 'RaptorX',\n",
       " 'Jmol',\n",
       " 'Ghemical',\n",
       " 'QGIS',\n",
       " 'LDAK',\n",
       " 'PrediXcan',\n",
       " 'Sleuth',\n",
       " 'MUSCLE',\n",
       " 'ClinProTools',\n",
       " 'BERT',\n",
       " 'PANTHER',\n",
       " 'TBLASTN',\n",
       " 'ranger',\n",
       " 'eggNOG-mapper',\n",
       " 'PostGIS',\n",
       " 'Ansys Fluent',\n",
       " 'T-Coffee',\n",
       " 'ProtTest',\n",
       " 'missForest',\n",
       " 'fgsea',\n",
       " 'CRAN',\n",
       " 'HINT',\n",
       " 'FoldIndex',\n",
       " 'GPS-SUMO',\n",
       " 'Zotero',\n",
       " 'rjags',\n",
       " 'Gaussian',\n",
       " 'LAMP',\n",
       " 'CellEngine',\n",
       " 'NONMEM',\n",
       " 'LSQMAN',\n",
       " 'Agilent Feature Extraction Software',\n",
       " 'Voronoia',\n",
       " 'EpiEstim',\n",
       " 'LoFreq',\n",
       " 'Circos',\n",
       " 'PyBEL',\n",
       " 'SIMCA',\n",
       " 'Boruta',\n",
       " 'Taxonomer',\n",
       " 'Joinpoint',\n",
       " 'AmiGO',\n",
       " 'iProClass',\n",
       " 'Image Quant TL',\n",
       " 'Phaser',\n",
       " 'ChromHMM',\n",
       " 'PHAST',\n",
       " 'Gorilla Experiment Builder',\n",
       " 'ProtVista',\n",
       " 'SMOG 2',\n",
       " 'PROMALS3D',\n",
       " 'TMHMM Server',\n",
       " 'SoupX',\n",
       " 'SCDE',\n",
       " 'NCBI',\n",
       " 'SMART',\n",
       " 'PHASE',\n",
       " 'DnaSP',\n",
       " 'Infernal',\n",
       " 'MIDAS',\n",
       " 'MIDAS',\n",
       " 'Tableau Desktop',\n",
       " 'RootNav',\n",
       " 'NIH Image',\n",
       " 'rtracklayer',\n",
       " 'IMPUTE2',\n",
       " 'Gblocks',\n",
       " 'TreeDyn',\n",
       " 'XCMS',\n",
       " 'FlowSOM',\n",
       " 'TranslatorX',\n",
       " 'Mapix',\n",
       " 'UpSet',\n",
       " 'Plotly',\n",
       " 'phyloseq',\n",
       " 'Cuffdiff',\n",
       " 'BLAT',\n",
       " 'JASPAR',\n",
       " 'readxl',\n",
       " 'Mcule',\n",
       " 'Argus',\n",
       " 'JWatcher',\n",
       " '3ds Max',\n",
       " 'Horos',\n",
       " 'PyCharm',\n",
       " 'Psychophysics Toolbox',\n",
       " 'PsychoPy',\n",
       " 'LigandScout',\n",
       " 'Cell-PLoc',\n",
       " 'TopHat',\n",
       " 'ChemRICH',\n",
       " 'VassarStats',\n",
       " 'FinchTV',\n",
       " 'iceLogo',\n",
       " 'Protein Prospector',\n",
       " 'Panoramic Viewer',\n",
       " 'CLC Main Workbench',\n",
       " 'PROVEAN',\n",
       " 'VisiView',\n",
       " 'MRIcron',\n",
       " 'HHpred',\n",
       " 'ProteinLynx Global Server',\n",
       " 'OMiCC',\n",
       " 'ArrayExpress',\n",
       " 'PROCESS',\n",
       " 'ORCA',\n",
       " 'PANDAseq',\n",
       " 'Segemehl',\n",
       " 'tRNAscan-SE',\n",
       " 'Short Time-series Expression Miner (STEM)',\n",
       " 'BrainVision Recorder',\n",
       " 'DiffBind',\n",
       " 'FoldX',\n",
       " 'LAMARC',\n",
       " 'TopSpin',\n",
       " 'TALOS-N',\n",
       " 'phytools',\n",
       " 'DoubletFinder',\n",
       " 'FAMHAP',\n",
       " 'CASA',\n",
       " 'ROSE',\n",
       " 'BRB-ArrayTools',\n",
       " 'NIS-Elements',\n",
       " 'SAINTexpress',\n",
       " 'PeptideProphet',\n",
       " 'ProteoWizard',\n",
       " 'ShinyGO',\n",
       " 'Neo',\n",
       " 'NetNGlyc',\n",
       " 'SOAPnuke',\n",
       " 'Metascape',\n",
       " 'MIRA',\n",
       " 'Biostrings',\n",
       " 'GLYCAM-Web',\n",
       " 'OpenSim',\n",
       " 'IMGT/HLA',\n",
       " 'SAVES',\n",
       " 'LigPlot+',\n",
       " 'Olympus',\n",
       " 'NetworkAnalyst',\n",
       " 'OpenEpi',\n",
       " 'Heatmapper',\n",
       " 'Canu',\n",
       " 'OligoCalc',\n",
       " 'NormFinder',\n",
       " 'GNPS',\n",
       " 'MSstats',\n",
       " 'CORUM',\n",
       " 'KnowEnG',\n",
       " 'mirTools',\n",
       " 'Buccaneer',\n",
       " 'PEAR',\n",
       " 'MiXCR',\n",
       " 'PyRosetta',\n",
       " 'Rosalind',\n",
       " 'RSeQC',\n",
       " 'WikiPathways',\n",
       " 'X!Tandem',\n",
       " 'GENEWIZ',\n",
       " 'PharmMapper',\n",
       " 'FlexX',\n",
       " 'Desmond',\n",
       " 'LDSC',\n",
       " 'Vensim',\n",
       " 'MacVector',\n",
       " 'MetaMap',\n",
       " 'MacroModel',\n",
       " 'NetPhos',\n",
       " 'Array Studio',\n",
       " 'Albacore',\n",
       " 'Gubbins',\n",
       " 'NanoFilt',\n",
       " 'PubMLST',\n",
       " 'CORAL',\n",
       " 'SeqKit',\n",
       " 'FlashPCA',\n",
       " 'Bash',\n",
       " 'Manubot',\n",
       " 'Leica Application Suite X',\n",
       " 'InterPro',\n",
       " 'SNAP',\n",
       " 'Phyre',\n",
       " 'WoLF PSORT',\n",
       " 'RACER',\n",
       " 'PhenStat',\n",
       " 'Genomic Ranges',\n",
       " 'DiscoTope',\n",
       " 'Ingenuity Pathway Analysis',\n",
       " 'ChAMP',\n",
       " 'ActiveDriver',\n",
       " 'Inkscape',\n",
       " 'Toxtree',\n",
       " 'ALOGPS',\n",
       " 'ConsensusClusterPlus',\n",
       " 'SplitsTree',\n",
       " 'Debian',\n",
       " 'Aspen Plus',\n",
       " 'Wolfram Mathematica',\n",
       " 'Swift',\n",
       " 'PatientsLikeMe',\n",
       " 'QualiMap',\n",
       " 'BisoGenet',\n",
       " 'QikProp',\n",
       " 'affyPLM',\n",
       " 'cowplot',\n",
       " 'TimeTree',\n",
       " 'Radiant',\n",
       " 'epiR',\n",
       " 'Phandango',\n",
       " 'FreeSurfer',\n",
       " 'ITK-SNAP',\n",
       " 'ITK-SNAP',\n",
       " 'FactoMineR',\n",
       " 'git-annex',\n",
       " 'ResearchGate',\n",
       " 'STEPS',\n",
       " 'MutaRNA',\n",
       " 'LifeGene',\n",
       " 'MS-GF+',\n",
       " 'EnhancedVolcano',\n",
       " 'ReactomePA',\n",
       " 'BIOVIA Discovery Studio',\n",
       " 'Google Charts',\n",
       " 'SigmaStat',\n",
       " 'MUMmer',\n",
       " 'ArrayOligoSelector',\n",
       " 'Open Targets Platform',\n",
       " 'PeptideAtlas',\n",
       " 'Offline Sorter',\n",
       " 'Newbler',\n",
       " 'DiANNA',\n",
       " 'AlgPred',\n",
       " 'riboWaltz',\n",
       " 'COBRApy',\n",
       " 'BioLayout Express 3D',\n",
       " 'ContigExpress',\n",
       " 'Strelka',\n",
       " 'AutoCAD',\n",
       " 'MEGA2',\n",
       " 'Partek Genomics Suite',\n",
       " 'openSMILE',\n",
       " 'Velocyto',\n",
       " 'SILVA',\n",
       " 'MongoDB',\n",
       " 'ProTox-II',\n",
       " 'WebPlotDigitizer',\n",
       " 'ShoRAH',\n",
       " 'FASTX-Toolkit',\n",
       " 'multcomp',\n",
       " 'Phobius',\n",
       " 'CYANA',\n",
       " 'Primer Designer',\n",
       " 'JCVI Primer Designer',\n",
       " 'trRosetta',\n",
       " 'GEOquery',\n",
       " 'robvis',\n",
       " 'EBSeq',\n",
       " 'viridis',\n",
       " 'Leginon',\n",
       " 'OpenMeta',\n",
       " 'Apache Tomcat',\n",
       " 'GNU Octave',\n",
       " 'ExpressionSuite',\n",
       " 'Preseq',\n",
       " 'BrainVision Analyzer',\n",
       " 'Hypermut',\n",
       " 'GlycoWorkbench',\n",
       " 'CellPhoneDB',\n",
       " 'GlyProt',\n",
       " 'IgBLAST',\n",
       " 'Terra',\n",
       " 'GAP',\n",
       " 'Agisoft Metashape',\n",
       " 'Nipype',\n",
       " 'fMRIPrep',\n",
       " 'ProteinPlus',\n",
       " 'AL2CO',\n",
       " 'minfi',\n",
       " 'tximport',\n",
       " 'Thermo Xcalibur',\n",
       " 'MetaCore',\n",
       " 'Adobe Photoshop Lightroom',\n",
       " 'PICRUSt',\n",
       " 'PICRUSt2',\n",
       " 'UCHIME',\n",
       " 'Babel',\n",
       " 'Kepler',\n",
       " 'bioKepler',\n",
       " 'ATSAS',\n",
       " 'CRISPOR',\n",
       " 'autoPROC',\n",
       " 'CheckM',\n",
       " 'FACS',\n",
       " 'CARP',\n",
       " 'QUAST',\n",
       " 'Wizard',\n",
       " 'YinOYang',\n",
       " 'TargetP',\n",
       " 'maSigPro',\n",
       " 'NEBcutter',\n",
       " 'piRNABank',\n",
       " 'Chemotaxis and Migration Tool',\n",
       " 'STAMP',\n",
       " 'WorldClim',\n",
       " 'SparCC',\n",
       " 'PicTar',\n",
       " 'IMGT/V-QUEST',\n",
       " 'QuickGO',\n",
       " 'BioXpress',\n",
       " 'CIBERSORT',\n",
       " 'Deeptools',\n",
       " 'MOPAC',\n",
       " 'Mercury',\n",
       " 'MSMS',\n",
       " 'Shiny',\n",
       " 'ExploreDTI',\n",
       " 'Geneshot',\n",
       " 'PhyloBayes',\n",
       " 'SciStat',\n",
       " 'UltraScan',\n",
       " 'Gatan Microscopy Suite',\n",
       " 'TMAP',\n",
       " 'PanoramaWeb',\n",
       " 'BioModels',\n",
       " 'CMA',\n",
       " 'ChimeraSlayer',\n",
       " 'SNPnexus',\n",
       " 'MAGMA',\n",
       " 'liftOver',\n",
       " 'gtsummary',\n",
       " 'Scrubber2',\n",
       " 'Noldus',\n",
       " 'Exomiser',\n",
       " 'APE',\n",
       " 'Phred',\n",
       " 'Phrap',\n",
       " 'OrthoFinder',\n",
       " 'BRIG',\n",
       " 'GeneCodis',\n",
       " 'CKAN',\n",
       " 'ImageScope',\n",
       " 'DrivAER',\n",
       " 'rpart',\n",
       " 'circlize',\n",
       " 'Ontobee',\n",
       " 'Open Ephys',\n",
       " 'BD CellQuest Pro',\n",
       " 'astropy',\n",
       " 'DisGeNET',\n",
       " 'DIA-NN',\n",
       " 'ViReMa',\n",
       " 'rentrez',\n",
       " 'Integrated',\n",
       " 'Pavian',\n",
       " 'Pathfinder',\n",
       " 'glmnet',\n",
       " 'Epson',\n",
       " 'SwissDock',\n",
       " 'Kraken',\n",
       " 'sctransform',\n",
       " 'OpenSesame',\n",
       " 'Bsoft',\n",
       " 'CHOPCHOP',\n",
       " 'Design-Expert',\n",
       " 'Spot',\n",
       " 'MSQuant',\n",
       " 'BioWorks',\n",
       " 'Slingshot',\n",
       " 'AUCell',\n",
       " 'UMI-tools',\n",
       " 'DEGseq',\n",
       " 'Zoomify',\n",
       " 'ModFit LT',\n",
       " 'Rsubread',\n",
       " 'HKL-2000',\n",
       " 'MotionCor2',\n",
       " 'RNA-SeQC',\n",
       " 'InterMine',\n",
       " 'Coursera',\n",
       " 'Velvet',\n",
       " 'Maq',\n",
       " 'Jaguar',\n",
       " 'NOTOCORD-hem',\n",
       " 'SphygmoCor',\n",
       " 'ACD/ChemSketch',\n",
       " 'PaDEL-Descriptor',\n",
       " 'FLASH',\n",
       " 'GenomicRanges',\n",
       " 'ChEA',\n",
       " 'NOISeq',\n",
       " 'baySeq',\n",
       " 'GSNAP',\n",
       " 'REDItools',\n",
       " 'ChromasPro',\n",
       " 'FFTW',\n",
       " 'Vector NTI',\n",
       " 'Cellpose',\n",
       " 'Cellpose',\n",
       " 'Leica Application Suite',\n",
       " 'CEAS',\n",
       " 'Signac',\n",
       " 'GenePattern',\n",
       " 'pydicom',\n",
       " 'Eelbrain',\n",
       " 'Prodigal',\n",
       " 'Prodigal',\n",
       " 'MetaGeneAnnotator',\n",
       " 'LDA',\n",
       " 'GapCloser',\n",
       " 'HaploReg',\n",
       " 'sLORETA',\n",
       " 'WinRHIZO',\n",
       " 'QUANTO',\n",
       " 'UniProtKB',\n",
       " 'Immune Epitope Database and Analysis Resource (IEDB)',\n",
       " 'MRtrix',\n",
       " 'KAnalyze',\n",
       " 'SAMstrt',\n",
       " 'SWAN',\n",
       " 'SGA',\n",
       " 'IDBA-UD',\n",
       " 'Neurolucida Explorer',\n",
       " 'PyMVPA',\n",
       " 'ERPLAB',\n",
       " 'BCILAB',\n",
       " 'GLIMMPSE',\n",
       " 'OMERO',\n",
       " 'CueMol',\n",
       " 'Trinotate',\n",
       " 'InfoStat',\n",
       " 'BLOCK-iT RNAi Designer',\n",
       " 'Cellosaurus',\n",
       " 'VecScreen',\n",
       " 'UniParc',\n",
       " 'PIR',\n",
       " 'Chromas',\n",
       " 'PeakRanger',\n",
       " 'CohortMethod',\n",
       " 'SnpSift',\n",
       " 'Benchling',\n",
       " 'BrainNet Viewer',\n",
       " 'PANDA',\n",
       " 'DeepCell',\n",
       " 'ADAPT',\n",
       " 'NEST Simulator',\n",
       " 'Elephant',\n",
       " 'Vue.js',\n",
       " 'CRISPRdirect',\n",
       " 'HCImage',\n",
       " 'VTK',\n",
       " 'TrackVis',\n",
       " 'Mauve',\n",
       " 'LASTZ',\n",
       " 'SOAPdenovo',\n",
       " 'Glimmer',\n",
       " 'TreeBeST',\n",
       " 'RepeatMasker',\n",
       " 'factoextra',\n",
       " 'OpenClinica',\n",
       " 'PISCES',\n",
       " 'ACT',\n",
       " 'GoSurfer',\n",
       " 'GeneDB',\n",
       " 'LCModel',\n",
       " 'BioPerl',\n",
       " 'RazerS',\n",
       " 'CUPSAT',\n",
       " 'ordinal',\n",
       " 'UALCAN',\n",
       " 'Raven',\n",
       " 'MuscleJ',\n",
       " 'Coot',\n",
       " 'Flowlogic',\n",
       " 'SeqPrep',\n",
       " 'statsmodel',\n",
       " 'AutoQuant',\n",
       " 'MetaboAnalystR',\n",
       " 'BiostaTGV',\n",
       " 'SnapGene Viewer',\n",
       " 'ENDOG',\n",
       " 'Composition Profiler',\n",
       " 'MetaVolcanoR',\n",
       " 'Illumina Real Time Analysis',\n",
       " 'PASTA',\n",
       " 'OligoArray',\n",
       " 'HUMAnN',\n",
       " 'PhyloPhlAn',\n",
       " 'GraPhlAn',\n",
       " 'psRNATarget',\n",
       " 'GenomeJack',\n",
       " 'Vienna RNA',\n",
       " 'Highcharts',\n",
       " 'PLEK',\n",
       " 'RSurveillance',\n",
       " 'Eagle',\n",
       " 'PubMatrix',\n",
       " 'ArrayTrack',\n",
       " 'UCSF ChimeraX',\n",
       " 'PhyDyn',\n",
       " 'GenStat',\n",
       " 'GlycoMod',\n",
       " 'Agisoft Photoscan',\n",
       " 'MRspa',\n",
       " 'NeuriteTracer',\n",
       " 'OmicsBean',\n",
       " 'SEDPHAT',\n",
       " 'PredictHaplo',\n",
       " 'scales',\n",
       " 'PyMC3',\n",
       " 'Multi Gauge',\n",
       " 'RegulomeDB',\n",
       " 'PRSice',\n",
       " 'STAPLE',\n",
       " 'Morpheus',\n",
       " '3D Slicer',\n",
       " 'QtiPlot',\n",
       " 'HTStream',\n",
       " 'Isaac',\n",
       " 'SNPRelate',\n",
       " 'RegRNA',\n",
       " 'CREST',\n",
       " 'Zooniverse',\n",
       " 'TomoPy',\n",
       " 'Pathway Tools',\n",
       " 'STARANISO',\n",
       " 'Visual Molecular Dynamics',\n",
       " 'R Project for Statistical Computing',\n",
       " 'Altmetric Explorer',\n",
       " 'LAMA',\n",
       " 'ReadCube',\n",
       " 'DeconSeq',\n",
       " 'MetAMOS',\n",
       " 'MetaVelvet',\n",
       " 'FastPCR',\n",
       " 'CPTAC',\n",
       " 'ANY-maze',\n",
       " 'Poretools',\n",
       " 'GeneWays',\n",
       " 'DEXSeq',\n",
       " 'SPICE',\n",
       " 'DADA',\n",
       " 'DHARMa',\n",
       " 'UNAFold',\n",
       " 'QUASR',\n",
       " 'MagicPlot',\n",
       " 'OSIRIS',\n",
       " 'reshape',\n",
       " 'devtools',\n",
       " 'DISCO',\n",
       " 'GASP',\n",
       " 'MG-RAST',\n",
       " 'BioCaster',\n",
       " 'Dynamo',\n",
       " 'MITRE',\n",
       " 'FileMaker',\n",
       " 'JMP Genomics',\n",
       " 'Monocle2',\n",
       " 'Scrublet',\n",
       " 'dmetar',\n",
       " 'MetaSite',\n",
       " 'Dialog',\n",
       " 'msigdbr',\n",
       " 'PSORT II',\n",
       " 'pRESTO',\n",
       " 'DNASIS',\n",
       " 'CRISP',\n",
       " 'PennCNV',\n",
       " 'SAMBLASTER',\n",
       " 'CCTOP',\n",
       " 'MobiDB',\n",
       " 'QI Macros',\n",
       " 'phantompeakqualtools',\n",
       " 'PIA',\n",
       " 'MAJIQ',\n",
       " 'Opera',\n",
       " 'MatrixDB',\n",
       " 'tweeDEseq',\n",
       " 'Molegro Virtual Docker',\n",
       " 'Ontologizer',\n",
       " 'Stereo Investigator',\n",
       " 'Stereo Investigator',\n",
       " 'SNVer',\n",
       " 'ObservableHQ',\n",
       " 'Karma',\n",
       " 'Interferome',\n",
       " 'GeneCards',\n",
       " 'Bandage',\n",
       " 'SqueezeMeta',\n",
       " 'Ray Meta',\n",
       " 'CAT',\n",
       " 'Ultimaker Cura',\n",
       " 'GeneMark',\n",
       " 'MDR',\n",
       " 'FastSNP',\n",
       " 'TFPGA',\n",
       " 'LeadIT',\n",
       " 'VIPERdb',\n",
       " 'MariaDB',\n",
       " 'HEM',\n",
       " 'Grinder',\n",
       " 'dbNSFP',\n",
       " 'MetaCyc',\n",
       " 'GAS',\n",
       " 'CLAIRE',\n",
       " 'CCP',\n",
       " 'Brainstorm',\n",
       " 'Orange Data Mining',\n",
       " 'CopyCaller',\n",
       " 'NetSurfP',\n",
       " 'OpenPose',\n",
       " 'DPABI',\n",
       " 'Protege',\n",
       " 'FASTSLINK',\n",
       " 'ImmuneSpace',\n",
       " 'flowWorkspace',\n",
       " 'VennDiagram',\n",
       " 'Ilastik',\n",
       " 'PredictSNP',\n",
       " 'mCSM',\n",
       " 'nsSNPAnalyzer',\n",
       " 'SINA',\n",
       " 'BaseSpace',\n",
       " 'Monocle3',\n",
       " 'CellChat',\n",
       " 'SeuratWrappers',\n",
       " 'ensembldb',\n",
       " 'Primer3Plus',\n",
       " 'Maxent',\n",
       " 'SURPI',\n",
       " 'A5-miseq',\n",
       " 'Metab',\n",
       " 'RankProd',\n",
       " 'SpoTyping',\n",
       " 'CVODE',\n",
       " 'NEURON',\n",
       " 'NEURON',\n",
       " 'GeneMarker',\n",
       " 'Online Kappa Calculator',\n",
       " 'Bismark',\n",
       " 'FastQ Screen',\n",
       " 'Bowtie 2',\n",
       " 'HaploClique',\n",
       " 'TCC',\n",
       " 'DGIdb',\n",
       " 'Globus Genomics',\n",
       " 'GeneSnap',\n",
       " 'richR',\n",
       " 'Clocklab',\n",
       " 'Mfuzz',\n",
       " 'CellTracker',\n",
       " 'Nexus',\n",
       " 'RNAsnp',\n",
       " 'TissueAtlas',\n",
       " 'BaSiC',\n",
       " 'Scion Image',\n",
       " 'SpeedSeq',\n",
       " 'SymPy',\n",
       " 'PharmGKB',\n",
       " 'SCENIC',\n",
       " 'GENIE',\n",
       " 'DELLY',\n",
       " 'Pindel',\n",
       " 'BreakSeq',\n",
       " 'HugeSeq',\n",
       " 'Tuxedo',\n",
       " 'OpenMS',\n",
       " 'MAPMAKER/EXP',\n",
       " 'PolySearch',\n",
       " 'PRICE',\n",
       " 'DataAssist',\n",
       " 'Cuffmerge',\n",
       " 'RAVE',\n",
       " 'Lasergene PrimerSelect',\n",
       " 'Autodesk Meshmixer',\n",
       " 'Dotter',\n",
       " 'Musite',\n",
       " 'RepeatModeler',\n",
       " 'proTRAC',\n",
       " 'LTR_retriever',\n",
       " 'BUSCO',\n",
       " 'Rtsne',\n",
       " 'Rtsne',\n",
       " 'CNVkit',\n",
       " 'AfterQC',\n",
       " 'Biofactoid',\n",
       " 'BindingDB',\n",
       " 'Textpresso',\n",
       " 'SuperDecisions',\n",
       " 'BioBERT',\n",
       " 'JuncBASE',\n",
       " ...]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matching"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6b8b7ffa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1952"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matching)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3ea44965",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "185147"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(submatch)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f51e0c12",
   "metadata": {},
   "outputs": [],
   "source": [
    "new_list = list(set(submatch))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "3b571304",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['rjags',\n",
       " 'Engine (',\n",
       " 'Biosystems QuantStudio',\n",
       " 'survival',\n",
       " 'istical',\n",
       " 'SNPSpD',\n",
       " 'BO',\n",
       " 'SCIP',\n",
       " 'Twist',\n",
       " 'Axio Scan.Z1',\n",
       " 'Neurobot',\n",
       " 'Refmac',\n",
       " 'Metafer',\n",
       " 'GPI',\n",
       " 'Protein-',\n",
       " 'ANO',\n",
       " 'SNPinfo Web Server',\n",
       " 'Calliope',\n",
       " 'ProtParam',\n",
       " 'cytoHubba',\n",
       " 'ModelDB',\n",
       " 'fMOST',\n",
       " 'LESYMAP',\n",
       " 'MIR',\n",
       " '++',\n",
       " 'NetOGly',\n",
       " 'Tak',\n",
       " 'OS X',\n",
       " 'RU',\n",
       " 'NanoFilt',\n",
       " 'ECV',\n",
       " 'Infernal',\n",
       " 'ABMapper',\n",
       " 'RASA',\n",
       " 'SomaDataIO',\n",
       " 'T2DiACoD',\n",
       " 'Inform',\n",
       " '3D Slicer',\n",
       " 'SAINT',\n",
       " 'GelQuant',\n",
       " 'sta',\n",
       " 'CARTA',\n",
       " 'HOMER',\n",
       " 'Effect',\n",
       " 'CRISPRdirect',\n",
       " 'POSSUM',\n",
       " 'CLUSTER',\n",
       " 'SitePainter',\n",
       " 'Automatic',\n",
       " 'SAMS',\n",
       " 'Bash',\n",
       " 'TMAT',\n",
       " 'DeepLabCut',\n",
       " 'SS',\n",
       " 'Maple',\n",
       " 'CUPSAT',\n",
       " 'SST',\n",
       " 'Open Targets Platform',\n",
       " 'Earth',\n",
       " 'MAM',\n",
       " 'Pipeline Pilot',\n",
       " 'iCount',\n",
       " 'DOT',\n",
       " 'Mw',\n",
       " 'doGlycans',\n",
       " 'Opt',\n",
       " 'SNPdat',\n",
       " 'PMI-',\n",
       " 'Adobe Audition',\n",
       " 'St',\n",
       " 'Quantum',\n",
       " 'Lux',\n",
       " 'System Toolbox',\n",
       " 'DTIPrep',\n",
       " 'PDB-Dev',\n",
       " 'field',\n",
       " 'PROMALS3D',\n",
       " 'MARS Data Analysis',\n",
       " 'GenomicRanges',\n",
       " 'insane',\n",
       " 'Prot',\n",
       " 'xiSEARCH',\n",
       " 'COMPAS',\n",
       " 'Deducer',\n",
       " 'HiTEC',\n",
       " 'SNPRelate',\n",
       " 'Poisson',\n",
       " 'ggrepel',\n",
       " 'PDF',\n",
       " 'ITC',\n",
       " 'Stat',\n",
       " 'PopLDdecay',\n",
       " 'Drishti',\n",
       " 'richR',\n",
       " 'TCAT',\n",
       " 'GI',\n",
       " 'Scrubber2',\n",
       " 'Single',\n",
       " 'XPS',\n",
       " 'WEB-based',\n",
       " 'Tagger',\n",
       " 'eLORETA',\n",
       " 'Analysis System',\n",
       " 'Numbers',\n",
       " 'iOS',\n",
       " 'APE',\n",
       " 'RPS',\n",
       " 'Synapse',\n",
       " 'Channel',\n",
       " 'Trace Archive',\n",
       " 'Open Babel',\n",
       " 'ANTIGENpro',\n",
       " 'PROMO',\n",
       " 'MOA',\n",
       " 'DEXSeq',\n",
       " '.js',\n",
       " 'MRIcron',\n",
       " 'Cell Counter',\n",
       " 'LDA',\n",
       " 'NetAffx',\n",
       " 'Generic GO Term Mapper',\n",
       " 'FAIRSharing',\n",
       " 'Zoomify',\n",
       " 'IncuCyte',\n",
       " 'Flow',\n",
       " 'AMA',\n",
       " 'software for',\n",
       " 'RDT',\n",
       " 'Band',\n",
       " 'Work',\n",
       " 'REDCap',\n",
       " 'PathXL',\n",
       " 'PALS',\n",
       " 'AW',\n",
       " 'CCC',\n",
       " 'Bamtools',\n",
       " 'Of',\n",
       " 'DOS',\n",
       " 'GraphPad P',\n",
       " 'frappe',\n",
       " 'Neuroscan',\n",
       " 'CADD',\n",
       " 'Proprietary',\n",
       " 'XI',\n",
       " 'Sco',\n",
       " 'NG',\n",
       " 'yn',\n",
       " 'Linux',\n",
       " 'Excel',\n",
       " 'NEBcutter',\n",
       " 'Apache Spark',\n",
       " 'KEGG',\n",
       " 'ExoCarta',\n",
       " 'Bambus',\n",
       " 'Google Scholar',\n",
       " 'BioSpace',\n",
       " 'NPInter',\n",
       " 'T5',\n",
       " 'Sequence Read Archive (SRA)',\n",
       " 'SC1',\n",
       " 'Mi',\n",
       " 'Phenograph',\n",
       " 'Time',\n",
       " 'pLabel',\n",
       " 'Immune Epitope Database and Analysis Resource (IEDB)',\n",
       " 'Plot Digitizer',\n",
       " 'DS)',\n",
       " 'SocioPatterns',\n",
       " '4D',\n",
       " 'Universal',\n",
       " 'PeakRanger',\n",
       " 'Code Ocean',\n",
       " 'Lisa',\n",
       " 'Sound Analysis Pro',\n",
       " 'DB2',\n",
       " 'GNU Image Manipulation Program',\n",
       " 'pgAdmin',\n",
       " 'QUARK',\n",
       " 'JWatcher',\n",
       " 'Prisma',\n",
       " 'Kit (',\n",
       " 'FlexProt',\n",
       " 'Clinical Terms',\n",
       " 'Name',\n",
       " 'Bsoft',\n",
       " 'ks',\n",
       " 'Sigma',\n",
       " 'Google Compute Engine',\n",
       " 'Symmetry',\n",
       " 'FASTER',\n",
       " 'GCBI',\n",
       " 'Roche',\n",
       " 'GraPhlAn',\n",
       " 'Mathematica',\n",
       " 'Recombination Detection',\n",
       " 'Venny',\n",
       " 'E300',\n",
       " 'ropls',\n",
       " 'Project for Statistical',\n",
       " 'ez',\n",
       " 'PS 2',\n",
       " 'Orbitrap Eclipse',\n",
       " 'USC',\n",
       " 'Singa',\n",
       " 'circlize',\n",
       " 'Silver',\n",
       " 'Ranch',\n",
       " 'CQN',\n",
       " 'TrackVis',\n",
       " 'QuickGO',\n",
       " 'ferred',\n",
       " 'SeqTrace',\n",
       " 'NetworkAnalyst',\n",
       " 'Node',\n",
       " 'T E',\n",
       " 'ConSurf',\n",
       " 'microT',\n",
       " 'TETRA',\n",
       " 'UCSC Xena',\n",
       " 'Ecology',\n",
       " 'Operetta',\n",
       " 'SeqMan Pro',\n",
       " 'MOLE',\n",
       " 'SciBite',\n",
       " 'Edge',\n",
       " 'QR',\n",
       " 'CD-',\n",
       " 'affyPLM',\n",
       " 'Cellpose',\n",
       " 'org',\n",
       " 'PyCharm',\n",
       " 'WT',\n",
       " 'Software System',\n",
       " 'VP',\n",
       " 'TREE',\n",
       " 'SIMM',\n",
       " 'GenAl',\n",
       " 'MIA',\n",
       " 'Miltenyi',\n",
       " 'Stem',\n",
       " 'Experiment Builder',\n",
       " 'Phandango',\n",
       " 'Cambridge',\n",
       " 'dows',\n",
       " 'Ensemble',\n",
       " 'devtools',\n",
       " 'agriGO',\n",
       " 'Pie',\n",
       " 'RCA2',\n",
       " 'VassarStats',\n",
       " 'Slicer',\n",
       " 'Rev',\n",
       " 'GM',\n",
       " 'T-REKS',\n",
       " 'piRNABank',\n",
       " 'Circlator',\n",
       " 'EIGENSTRAT',\n",
       " 'Click',\n",
       " 'GAIN',\n",
       " 'SCS',\n",
       " 'ACT',\n",
       " 'RK',\n",
       " 'Windows',\n",
       " 'MA-',\n",
       " 'DNA R',\n",
       " 'MIX',\n",
       " 'Dimension',\n",
       " 'Public',\n",
       " 'ua',\n",
       " 'Meeting',\n",
       " 'STAMP',\n",
       " 'VMD',\n",
       " 'Variant',\n",
       " 'Jmol',\n",
       " 'Ger',\n",
       " 'Handbook',\n",
       " 'Adobe Photoshop Lightroom',\n",
       " 'eggNOG-',\n",
       " 'QuRe',\n",
       " 'derived',\n",
       " 'tran',\n",
       " 'AF)',\n",
       " 'Inv',\n",
       " 'GENECLASS',\n",
       " 'SURPI',\n",
       " 'Connect',\n",
       " 'PMI',\n",
       " 'PAA',\n",
       " 'Amersham Imager',\n",
       " 'CyVerse',\n",
       " 'SDP',\n",
       " 'imzML',\n",
       " 'Ultimate',\n",
       " 'SCV',\n",
       " 'Glyco3',\n",
       " 'SPICE',\n",
       " 'MoonProt',\n",
       " 'AUCell',\n",
       " 'RaptorX',\n",
       " 'Origin',\n",
       " 'cro',\n",
       " 'Gnuplot',\n",
       " 'martinize.py',\n",
       " 'JG',\n",
       " 'FP',\n",
       " 'Bioclipse',\n",
       " 'A5',\n",
       " 'Director',\n",
       " 'GHOST',\n",
       " 'BiofilmQ',\n",
       " 'clusterPro',\n",
       " 'higher',\n",
       " 'GTDB-Tk',\n",
       " 'Phyutility',\n",
       " 'BBD',\n",
       " 'PSA',\n",
       " 'lme4',\n",
       " 'Maestro',\n",
       " 'XNAT',\n",
       " 'tidyr',\n",
       " 'DADA',\n",
       " 'Karma',\n",
       " 'MISO',\n",
       " 'RStu',\n",
       " 'Fam',\n",
       " 'survminer',\n",
       " 'STATISTI',\n",
       " 'ASTRA',\n",
       " 'Agisoft Metashape',\n",
       " 'Locator',\n",
       " 'XN',\n",
       " 'CEA',\n",
       " 'Google Sites',\n",
       " 'R R',\n",
       " 'Polymorphism Phenotyping',\n",
       " 'Platform',\n",
       " 'CompuCell3D',\n",
       " 'pyDNase',\n",
       " 'DAAC',\n",
       " 'GenomeBrowse',\n",
       " 'SpoTyping',\n",
       " 'canSAR',\n",
       " 'DYNA',\n",
       " 'Ringo',\n",
       " 'BSD',\n",
       " 'CleanLine',\n",
       " 'GlycanBuilder',\n",
       " 'T200',\n",
       " 'TIES',\n",
       " 'SuperPred',\n",
       " 'Phan',\n",
       " 'FT',\n",
       " 'Sequence Detection System',\n",
       " 'Globus',\n",
       " 'GNU Octave',\n",
       " 'Pages',\n",
       " 'Application',\n",
       " 'ROB',\n",
       " 'SingleCell',\n",
       " 'MetAMOS',\n",
       " 'NIRS-SPM',\n",
       " 'RCSB',\n",
       " 'TinderMIX',\n",
       " 'GeneTools',\n",
       " 'Exomiser',\n",
       " 'Kepler',\n",
       " 'XPLOR',\n",
       " 'Ali',\n",
       " 'CMS',\n",
       " 'NVivo',\n",
       " 'Schro',\n",
       " 'AngioCalc',\n",
       " 'SMART',\n",
       " 'GARD',\n",
       " 'MToolBox',\n",
       " 'R tool',\n",
       " 'GlyPro',\n",
       " 'Scout',\n",
       " 'SocioPattern',\n",
       " 'Nim',\n",
       " 'CIPHER',\n",
       " 'AMBIT',\n",
       " 'RAP-DB',\n",
       " 'ACY',\n",
       " 'Simula',\n",
       " 'Utopia',\n",
       " 'Mongo',\n",
       " 'LMAP',\n",
       " 'Monster',\n",
       " 'FOM',\n",
       " 'MeGO',\n",
       " 'AML',\n",
       " '3Omics',\n",
       " 'DnaSP',\n",
       " 'Proteome Discoverer',\n",
       " 'IVD',\n",
       " 'tRNAscan-SE',\n",
       " 'Tumor',\n",
       " 'ABAS',\n",
       " 'yStat',\n",
       " 'ArchR',\n",
       " 'CP',\n",
       " 'Tel',\n",
       " 'Iris',\n",
       " 'LC',\n",
       " 'PriVar',\n",
       " 'Thermo Xcalibur',\n",
       " 'galore',\n",
       " 'IQTL',\n",
       " 'Proteome Discover',\n",
       " 'Apache Hadoop',\n",
       " 'UC',\n",
       " 'DEC',\n",
       " 'GAP',\n",
       " 'TEQC',\n",
       " 'Ensembl Fungi',\n",
       " 'CX',\n",
       " 'Selector',\n",
       " 'ImageScope',\n",
       " 'Pack',\n",
       " 'PRIDE',\n",
       " 'IVS',\n",
       " 'HAP',\n",
       " 'Open',\n",
       " 'NGA',\n",
       " 'Genomics Suite',\n",
       " 'Binder',\n",
       " 'MID',\n",
       " 'SeqMan',\n",
       " 'gmap',\n",
       " 'MPP',\n",
       " 'HapCompass',\n",
       " 'Keypoint',\n",
       " 'PhenStat',\n",
       " 'PE',\n",
       " 'http',\n",
       " 'RST',\n",
       " 'riboWaltz',\n",
       " 'ensembldb',\n",
       " 'HAPMIX',\n",
       " 'Autogrid',\n",
       " 'React',\n",
       " 'Seq2',\n",
       " 'SICER',\n",
       " 'pip',\n",
       " 'Phyre',\n",
       " 'SEA',\n",
       " 'BIA',\n",
       " 'featureCounts',\n",
       " 'irr',\n",
       " 'Finder',\n",
       " 'Genboree Workbench',\n",
       " 'Random DNA Sequence Generator',\n",
       " 'TopScan Lite',\n",
       " 'PANDAseq',\n",
       " 'S4',\n",
       " 'Jupyter Notebook',\n",
       " 'PK',\n",
       " 'Sequence Read Archive (SRA',\n",
       " 'meta',\n",
       " 'taxize',\n",
       " 'GREG',\n",
       " 'DISULFIND',\n",
       " 'MIM',\n",
       " 'R Core',\n",
       " 'Helicon Focus',\n",
       " 'STRUCTURE',\n",
       " 'Array Analyzer',\n",
       " 'Phospho.ELM',\n",
       " 'M 2',\n",
       " 'BeadArray',\n",
       " 'VAM',\n",
       " 'Nor',\n",
       " 'MacroModel',\n",
       " 'GENIES',\n",
       " 'Lem',\n",
       " 'Infinity',\n",
       " 'Score',\n",
       " 'NES',\n",
       " 'MSA',\n",
       " 'CYT',\n",
       " 'GERP',\n",
       " 'ERAN',\n",
       " 'StarDrop',\n",
       " 'Profile',\n",
       " 'MetaMapR',\n",
       " 'ICE',\n",
       " 'Ensembles',\n",
       " 'SDS',\n",
       " 'Conservation',\n",
       " 'Genedata Screener',\n",
       " 'MolProbity',\n",
       " 'STAPLE',\n",
       " 'XGBoost',\n",
       " 'CODE',\n",
       " 'GIM',\n",
       " 'MZmine',\n",
       " 'JMP',\n",
       " 'Prabi',\n",
       " 'Hori',\n",
       " 'GPC/SEC',\n",
       " 'DNACLUST',\n",
       " 'Win',\n",
       " 'DeepL',\n",
       " 'ChemBio',\n",
       " 'BUSTER',\n",
       " 'EASE',\n",
       " 'SAT-',\n",
       " 'ECOFFinder',\n",
       " 'ENA',\n",
       " 'NPTool',\n",
       " 'Biorender',\n",
       " 'MSO',\n",
       " 'Topol',\n",
       " 'Echo',\n",
       " 'AdapterRemoval',\n",
       " 'nCounter',\n",
       " 'SHRINE',\n",
       " 'glmm',\n",
       " 'FlowSim',\n",
       " 'BloodSpot',\n",
       " 'Sleuth',\n",
       " 'ConceptMapper',\n",
       " 'Ingenuity Pathway',\n",
       " 'GeSeq',\n",
       " 'Jupyter',\n",
       " 'Warehouse',\n",
       " 'Light Cycler 480',\n",
       " 'itope',\n",
       " 'ExpressionSuite',\n",
       " 'Singularity',\n",
       " 'YMDB',\n",
       " 'Mapix',\n",
       " 'BioPlex',\n",
       " 'slam',\n",
       " 'Recombination Detection Program',\n",
       " 'PAT',\n",
       " 'Abs',\n",
       " 'Array Express',\n",
       " 'gini',\n",
       " 'CCH',\n",
       " 'PATH',\n",
       " 'GeneStudio',\n",
       " 'MiSeq FGx',\n",
       " 'ImmuneSpace',\n",
       " 'Applied Biosystems',\n",
       " 'Pro',\n",
       " 'Confluence',\n",
       " 'Cytation 5',\n",
       " 'Interaction Map',\n",
       " 'ImaGene',\n",
       " 'Clus',\n",
       " 'Bedside',\n",
       " 'R-3',\n",
       " 'VIPERdb',\n",
       " 'Image Lab',\n",
       " 'Cas',\n",
       " 'Amazon',\n",
       " 'dbSNP',\n",
       " 'CNVkit',\n",
       " 'Mfuzz',\n",
       " 'Mr',\n",
       " 'SIMCA',\n",
       " 'Harmony',\n",
       " 'EZ-C1',\n",
       " 'MicroCosm',\n",
       " 'Vision',\n",
       " 'Drop-seq tools',\n",
       " 'MAC',\n",
       " 'ShortStack',\n",
       " 'Zeiss',\n",
       " 'Phylo',\n",
       " 'SSE',\n",
       " 'statis',\n",
       " 'CU',\n",
       " 'Jude',\n",
       " 'IIS',\n",
       " 'dChip',\n",
       " 'CRUISE',\n",
       " 'GAMMA',\n",
       " 'TissueQuest',\n",
       " 'InterMine',\n",
       " 'Geisinger',\n",
       " 'FLIMfit',\n",
       " 'FlowCal',\n",
       " 'retriever',\n",
       " 'CDC',\n",
       " 'SparCC',\n",
       " 'UMass',\n",
       " 'L 1',\n",
       " 'Merlin',\n",
       " 'EndNote',\n",
       " 'PyRx',\n",
       " 'RA',\n",
       " 'pFind',\n",
       " 'PINA',\n",
       " 'LAMP',\n",
       " 'Scion Image',\n",
       " 'VISTA Browser',\n",
       " 'ECAT',\n",
       " 'NSP',\n",
       " 'Vaa3D',\n",
       " 'Image-Pro Express',\n",
       " 'YASARA',\n",
       " 'NeuroVault',\n",
       " 'Mu',\n",
       " 'VA',\n",
       " 'Magellan',\n",
       " 'MCR',\n",
       " 'DeepPoseKit',\n",
       " 'REP',\n",
       " 'Just',\n",
       " 'X9',\n",
       " 'sys',\n",
       " 'TPL',\n",
       " 'Mycobrowser',\n",
       " 'PDB Validation Server',\n",
       " 'BUGS',\n",
       " 'PhotoCapt',\n",
       " 'Cori',\n",
       " 'Aline',\n",
       " 'Tracer',\n",
       " 'DICAL',\n",
       " 'Ord',\n",
       " 'NEIL',\n",
       " 'MOSES',\n",
       " 'SciStat',\n",
       " 'GeneAtlas',\n",
       " 'MiRP',\n",
       " 'RepeatModeler',\n",
       " 'Align-GVGD',\n",
       " 'Tissue',\n",
       " 'Aquaria',\n",
       " 'Grinder',\n",
       " 'ImageMagick',\n",
       " 'Bridger',\n",
       " 'ImageX',\n",
       " 'eMovie',\n",
       " 'CAMA',\n",
       " 'QL',\n",
       " 'PMINE',\n",
       " 'WTDBG',\n",
       " 'Black Zen',\n",
       " 'KD',\n",
       " 'Trinotate',\n",
       " 'mma',\n",
       " 'cryoSPARC',\n",
       " 's Toolbox',\n",
       " 'CRI',\n",
       " 'Uni',\n",
       " 'RNA FRABASE',\n",
       " 'RAMPAGE',\n",
       " 'epiR',\n",
       " 'CDD',\n",
       " 'ChIPseeker',\n",
       " 'PatientsLikeMe',\n",
       " 'BLAT',\n",
       " 'TBLASTX',\n",
       " 'Snakemake',\n",
       " 'ABSOLUTE',\n",
       " 'pec',\n",
       " 'ages',\n",
       " 'Schrodinger',\n",
       " 'PISA',\n",
       " 'Programme',\n",
       " 'MiST',\n",
       " 'FACS Sortware',\n",
       " 'Together',\n",
       " 'RootNav',\n",
       " 'CellProfiler',\n",
       " 'Recognition',\n",
       " 'BepiPred-2',\n",
       " 'Clinic',\n",
       " 'Hex',\n",
       " 'htseq-count',\n",
       " 'NormFinder',\n",
       " 'XPRESS',\n",
       " 'RC',\n",
       " 'BrainVision Recorder',\n",
       " 'Eprime',\n",
       " 'Tabula',\n",
       " 'Areas',\n",
       " 'Warp',\n",
       " 'PASTA',\n",
       " 'cron',\n",
       " 'Acuity',\n",
       " 'EasyView',\n",
       " 'Wolfram Mathematica',\n",
       " 'ContentMine',\n",
       " 'loPhlAn',\n",
       " 'CASS',\n",
       " 'MACSQuantify',\n",
       " 'G.',\n",
       " 'Zoom',\n",
       " 'Kaluza',\n",
       " 'ShortRead',\n",
       " 'Translate',\n",
       " 'SolexaQA',\n",
       " 'Agilent MassHunter Qualitative Analysis',\n",
       " 'LINKAGE',\n",
       " 'Di',\n",
       " 'Using',\n",
       " 'C50',\n",
       " 'TCC',\n",
       " 'Turk',\n",
       " 'RME',\n",
       " 'Imaging and',\n",
       " 'PRIME',\n",
       " 'LSM 510',\n",
       " 'BayeScan',\n",
       " 'Google Charts',\n",
       " 'DCC',\n",
       " 'Anton',\n",
       " 'Cindy',\n",
       " 'ntbox',\n",
       " 'E-CRISP',\n",
       " 'Model',\n",
       " 'EIT',\n",
       " 'MITK',\n",
       " 'Biology',\n",
       " 'plyr',\n",
       " 'HumanNet',\n",
       " 'r',\n",
       " 'ADAPT',\n",
       " 'Dialog',\n",
       " 'ProteinLynx Global Server',\n",
       " 'ViralZone',\n",
       " 'DEP',\n",
       " '.Hs.eg',\n",
       " 'Meta Server',\n",
       " 'SerialEM',\n",
       " 'PEAK',\n",
       " 'UMAP',\n",
       " 'INA',\n",
       " 'Membrane',\n",
       " 'tbl2asn',\n",
       " 'Segment',\n",
       " 'ASTRAL',\n",
       " '3d',\n",
       " 'Cellosaurus',\n",
       " 'CM',\n",
       " 'in FSL',\n",
       " 'CyTOF',\n",
       " 'START',\n",
       " 'Ver',\n",
       " 'OneLab',\n",
       " 'TRACESPipe',\n",
       " 'Source',\n",
       " 'Request',\n",
       " 'REL',\n",
       " 'RazerS',\n",
       " 'minfi',\n",
       " 'afex',\n",
       " 'FASTX-Toolkit',\n",
       " 'GDA',\n",
       " 'CDF',\n",
       " 'instrument',\n",
       " 'Bing',\n",
       " 'Assign',\n",
       " 'RIN',\n",
       " 'Sage',\n",
       " 'ICU',\n",
       " 'Tips',\n",
       " 'Mutation Surveyor',\n",
       " 'SEDPHAT',\n",
       " 'CAS Medical',\n",
       " 'Event',\n",
       " 'FACSAria',\n",
       " 'Mobius',\n",
       " 'Pavian',\n",
       " 'Rdisop',\n",
       " 'Animal',\n",
       " 'sim',\n",
       " 'Mcule',\n",
       " 'for the',\n",
       " 'Agilent Feature',\n",
       " 'GtRNAdb',\n",
       " 'Google Sheet',\n",
       " 'Science',\n",
       " '3ds Max',\n",
       " 'miR-',\n",
       " 'TERA',\n",
       " 'fig',\n",
       " 'IMGT/StatClonotype',\n",
       " 'ORC',\n",
       " 'caret',\n",
       " 'FSL',\n",
       " 'Vital',\n",
       " 'git-annex',\n",
       " 'ExomeDepth',\n",
       " 'BioModels',\n",
       " 'Statistic',\n",
       " 'zoo',\n",
       " 'FreeSurfer',\n",
       " 'Che',\n",
       " 'Design',\n",
       " 'rsync',\n",
       " 'Foundation for',\n",
       " 'MRtrix',\n",
       " 'SMI BeGaze',\n",
       " 'FS',\n",
       " 'Helix',\n",
       " 'Hs.eg.db',\n",
       " 'Surveyor',\n",
       " 'OMERO',\n",
       " 'AAC',\n",
       " 'HiC-Pro',\n",
       " 'PSI',\n",
       " 'eTools',\n",
       " 'Epson',\n",
       " 'RGI',\n",
       " 'R-tool',\n",
       " 'Progenesis QI',\n",
       " 'FMP',\n",
       " 'NetMH',\n",
       " 'PMOD',\n",
       " 'soft',\n",
       " 'Gen',\n",
       " 'Blue',\n",
       " 'Class',\n",
       " 'Video Freeze',\n",
       " 'CRISPOR',\n",
       " 'MITRE',\n",
       " 'Sig',\n",
       " 'MALT',\n",
       " 'Preseq',\n",
       " 'TLA',\n",
       " 'MEGAN',\n",
       " 'HOPE',\n",
       " 'Statulator',\n",
       " 'FIRST',\n",
       " 'PICRUSt2',\n",
       " 'C4',\n",
       " 'iRmap',\n",
       " 'SeuratWrappers',\n",
       " 'Viewer',\n",
       " 'BD FACS',\n",
       " 'gle',\n",
       " 'Simpl',\n",
       " 'Agisoft Photoscan',\n",
       " 'GOR',\n",
       " 'Zotero',\n",
       " 'Applied Biosystem',\n",
       " 'ters',\n",
       " 'MOE',\n",
       " 'Brain',\n",
       " 'Berkeley',\n",
       " 'CNS',\n",
       " 'Scala',\n",
       " 'TAB',\n",
       " 'GPC',\n",
       " 'R package',\n",
       " 'Gblocks',\n",
       " 'fd',\n",
       " 'Short Time-series Expression Miner (STEM)',\n",
       " 'SPSS Amos',\n",
       " 'Blast2',\n",
       " 'Pegas',\n",
       " 'TargetP',\n",
       " 'PROCESS',\n",
       " 'Everest',\n",
       " 'Cypher',\n",
       " 'PHEM',\n",
       " 'biomaRt',\n",
       " 'TomoPy',\n",
       " 'SAMstrt',\n",
       " 'Calc',\n",
       " 'SYBYL',\n",
       " 'Extraction',\n",
       " 'InParanoid',\n",
       " 'Inferelator',\n",
       " 'Digital',\n",
       " 'ACS',\n",
       " 'Look',\n",
       " 'ngs.plot',\n",
       " 'MetaFluor',\n",
       " 'FUN',\n",
       " 'Max',\n",
       " 'Vina',\n",
       " 'Data Management System',\n",
       " 'HCImage',\n",
       " 'PASS',\n",
       " 'IMS',\n",
       " 'SNIP',\n",
       " 'ground',\n",
       " 'Measures of Effect Size Toolbox',\n",
       " 'Najmanovich',\n",
       " 'Sheets',\n",
       " 'Nipype',\n",
       " 'Spacer',\n",
       " 'PI',\n",
       " 'Real Time PCR',\n",
       " 'Cython',\n",
       " 'StatPac',\n",
       " 'PLS',\n",
       " 'Fluoview',\n",
       " 'MicrobiomeAnalyst',\n",
       " 'MobiDB',\n",
       " 'Population',\n",
       " 'IBM',\n",
       " 'Noldus',\n",
       " 'GeneMark',\n",
       " 'UK',\n",
       " 'Chinese',\n",
       " 'THOR',\n",
       " 'FAT',\n",
       " 'Reaction Kinetic',\n",
       " 'VFDB',\n",
       " 'PredictHaplo',\n",
       " 'MicroVigene',\n",
       " 'Constrain',\n",
       " 'ENIGMA-DTI',\n",
       " 'surveillance',\n",
       " 'Opticon Monitor',\n",
       " 'Express',\n",
       " 'CWL',\n",
       " 'T3',\n",
       " 'SSP',\n",
       " 'DISCO',\n",
       " 'Corti',\n",
       " 'IDBA',\n",
       " 'Phaser',\n",
       " 'Agora',\n",
       " 'Classification',\n",
       " 'metric',\n",
       " 'CellPro',\n",
       " 'MRD',\n",
       " 'Fluidigm C',\n",
       " 'PS',\n",
       " 'SCIRun',\n",
       " 'Models',\n",
       " 'C',\n",
       " 'Pin',\n",
       " 'Gel',\n",
       " 'covar',\n",
       " 'CoV-2',\n",
       " 'AS',\n",
       " 'AIML',\n",
       " 'Pocket',\n",
       " 'Development Kit',\n",
       " 'VDJ',\n",
       " 'Solomon Coder',\n",
       " 'Primer Express',\n",
       " 'Circos',\n",
       " 'ROCESS',\n",
       " 'ming',\n",
       " 'SDT',\n",
       " 'QS',\n",
       " 'FIVE',\n",
       " 'CueMol',\n",
       " 'Lab',\n",
       " 'JTK_CYCLE',\n",
       " 'ADT',\n",
       " 'core',\n",
       " 'ROMI',\n",
       " 'Detective',\n",
       " 'Ime',\n",
       " 'Effect Predictor',\n",
       " 'edit',\n",
       " 'ana-',\n",
       " 'Analyzers',\n",
       " 'FLASH',\n",
       " 'GeneCard',\n",
       " 'DM',\n",
       " 'OSM',\n",
       " 'immuneML',\n",
       " 'ASPicDB',\n",
       " 'SBP',\n",
       " 'SCC',\n",
       " 'Meteor',\n",
       " 'Glide',\n",
       " 'Federated',\n",
       " 'OligoA',\n",
       " 'Beacon',\n",
       " 'EMMA',\n",
       " 'mzCloud',\n",
       " 'Trimmomatic',\n",
       " 'loo',\n",
       " 'Drop',\n",
       " 'MUMmer',\n",
       " 'OSIRIS',\n",
       " 'straints',\n",
       " 'dBase',\n",
       " 'SQL',\n",
       " 'xCELLigence RTCA',\n",
       " 'MedScan',\n",
       " 'Similarity Analysis',\n",
       " 'BreakSeq',\n",
       " 'SMOG 2',\n",
       " 'ASET',\n",
       " 'GA',\n",
       " 'Boruta',\n",
       " 'Multi Gauge',\n",
       " 'AcqKnowledge',\n",
       " 'MHC',\n",
       " 'Combinatorial Extension (CE)',\n",
       " 'NGS',\n",
       " 'CANDID',\n",
       " 'RobotReviewer',\n",
       " 'Genedata Analyst',\n",
       " ...]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9c7a952b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5383"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f2fdc7b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"submatcheds.txt\", \"w\", encoding=\"utf-8\") as f:\n",
    "    f.write('\\n'.join(new_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0b473f80",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
